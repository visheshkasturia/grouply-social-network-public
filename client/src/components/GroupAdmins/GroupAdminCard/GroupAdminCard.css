.admin-card-container {
  display: grid;
  grid-template-columns: 10% 60%  30%;
  height: 75px;
  border: solid 3px black;
  border-radius: 10px;
  margin: 15px;
}

.admin-card-SNo {
  grid-column: 1/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 24px;
  font-weight: 400;
}

.admin-card-name {
  grid-column: 2/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 24px;
  font-weight: 400;
}


.admin-card-demoteAdmin {
  grid-column: 3/ span 1;
  display: flex;
  justify-content: center;
  align-items: center;
}
