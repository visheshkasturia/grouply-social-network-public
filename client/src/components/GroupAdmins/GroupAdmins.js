import React, { useState, useEffect } from 'react';
import AdminCard from './GroupAdminCard/GroupAdminCard';

/* eslint-disable */
function GroupAdmins({ user, group }) {
  const [groupAdminList, setGroupAdminList] = useState(group.admins)

  const getChangedAdminList = (data) => {
    setGroupAdminList(data);
  } 

  useEffect(() => {
    console.log(groupAdminList)
  }, [groupAdminList])
  return (
    <div>
      {
        groupAdminList.map((admin, i) => (
          <AdminCard key={admin} getChangedAdminList={getChangedAdminList} admin={admin} user={user} i={i + 1} group={group} />
        ))
      }
    </div>
  )
}

export default GroupAdmins;
