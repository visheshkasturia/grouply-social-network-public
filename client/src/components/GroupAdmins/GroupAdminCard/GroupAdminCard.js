import React from 'react';
import './GroupAdminCard.css';
import { demoteAdminInGroup } from '../../../apiCalls/demoteAdminInGroup';
/* eslint-disable */

function AdminCard({ admin, user, i, getChangedAdminList, group }) {

  const demoteAdmin = () => {
    demoteAdminInGroup(admin, user._id, group._id)
      .then((data) => {
        if (data.message == 'Success') {
          getChangedAdminList(data.res.admins);
        }
      })
      .catch((err) => {
        console.log(err)
      });
    return
  }

  return (
    <div className="admin-card-container">
      <div className="admin-card-SNo flex-center"> {i} </div>
      <div className="admin-card-name flex-center"> {admin} </div>
      <div className="admin-card-demoteAdmin flex-center">
      <button type="button" onClick={demoteAdmin}>Demote Admin</button>
      </div>  
    </div>
  )
}

export default AdminCard;
