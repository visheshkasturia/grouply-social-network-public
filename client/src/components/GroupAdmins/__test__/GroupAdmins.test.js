import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import GroupAdmins from '../GroupAdmins'
import GroupAdminCard from '../GroupAdminCard/GroupAdminCard'

const admin = 'abc'
const group = {
  groupName: "test",
  description: "desc",
  admins: [admin]
}

it('GroupAdmins renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<GroupAdmins group={group} />, div);
});

it('GroupAdmins matches snapshot', () => {
  const tree = renderer.create(<GroupAdmins group={group} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('GroupAdminCard renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<GroupAdminCard group={group} admin={admin} />, div);
});

it('GroupAdminCard matches snapshot', () => {
  const tree = renderer.create(<GroupAdminCard admin={admin} />).toJSON();
  expect(tree).toMatchSnapshot();
});