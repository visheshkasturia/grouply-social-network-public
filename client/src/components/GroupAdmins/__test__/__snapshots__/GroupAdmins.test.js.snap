// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupAdminCard matches snapshot 1`] = `
<div
  className="admin-card-container"
>
  <div
    className="admin-card-SNo flex-center"
  >
     
     
  </div>
  <div
    className="admin-card-name flex-center"
  >
     
    abc
     
  </div>
  <div
    className="admin-card-demoteAdmin flex-center"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Demote Admin
    </button>
  </div>
</div>
`;

exports[`GroupAdmins matches snapshot 1`] = `
<div>
  <div
    className="admin-card-container"
  >
    <div
      className="admin-card-SNo flex-center"
    >
       
      1
       
    </div>
    <div
      className="admin-card-name flex-center"
    >
       
      abc
       
    </div>
    <div
      className="admin-card-demoteAdmin flex-center"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Demote Admin
      </button>
    </div>
  </div>
</div>
`;
